src/foam/nanos/session/Session.js

```diff
@@ -243,6 +247,59 @@ foam.CLASS({
           .put("group", group)
           .put("appConfig", group.getAppConfig(rtn));
       `
+    },
+    {
+      name: 'validate',
+      args: [
+        { name: 'x', type: 'Context' }
+      ],
+      javaCode: `
+        if ( getUserId() < 0 ) {
+          throw new IllegalStateException("User id is invalid.");
+        }
+
+        if ( getAgentId() < 0 ) {
+          throw new IllegalStateException("Agent id is invalid.");
+        }
+
+        if ( getUserId() > 0 ) {
+          checkUserEnabled(x, getUserId());
+        }
+
+        if ( getAgentId() > 0 ) {
+          checkUserEnabled(x, getAgentId());
+
+          DAO agentJunctionDAO = (DAO) x.get("agentJunctionDAO");
+          UserUserJunction junction = (UserUserJunction) agentJunctionDAO.find(
+            AND(
+              EQ(UserUserJunction.SOURCE_ID, getAgentId()),
+              EQ(UserUserJunction.TARGET_ID, getUserId())
```

Comment:
```suggestion
              EQ(UserUserJunction.TARGET_ID, getUserId()),
              EQ(UserUserJunction.STATUS, net.nanopay.auth.AgentJunctionStatus.ACTIVE)
```

Reply:
net.nanopay.auth.AgentJunctionStatus is currently on nanopay side.

Reply:
I took note of this and we'll handle this separately. Might move that feature to FOAM or drop it.